attachments/forms.py

```diff
@@ -59,9 +59,9 @@ def __init__(self, *args, **kwargs):
         for prop in qs:
             if is_upload:
                 field_key = 'upload-%d-%s' % (instance.pk, prop.slug)
-                self.fields[field_key] = self.formfield(prop, 
-                                                        initial=form_data.get(field_key, None))
-            elif is_attachment:
+                self.fields[field_key] = self.formfield(prop,
+                                                        initial=instance.session.data.get(field_key, None) if instance.session.data else None)
+            elif isinstance(instance, Attachment):
```

Comment:
I think these changes should use what was already on 3.0 (form_data instead of instance.session.data, is_attachment instead of isinstance).

Reply:
Agreed, changed in my branch (MR coming soon)